app/extensions/safe/server-routes/safe.js

```diff
@@ -19,19 +25,60 @@ const safeRoute = {
                 return reply( 'SAFE not connected yet' );
             }
 
-            const data = await app.webFetch( link );
+            if ( headers.range )
+            {
+                const range = headers.range;
+                const rangeArray =
+                  range.substring( BYTES.length, range.length )
+                      .replace( /['"]+/g, '' )
+                      .split( '-' );
+
+                start = rangeArray[0] ? parseInt( rangeArray[0], 10 ) : null;
+                end = rangeArray[1] ? parseInt( rangeArray[1], 10 ) : null;
+
+                //we need to separate out 0- length requests which are not partial content
+                if( start && start !== 0 )
+                {
+                    //should have a start !== 0
+                    isRangeReq = true;
+                }
+
+                if( start === 0 && end )
+                {
+                    isRangeReq = true;
+                }
+            }
+
+            // setup opts object
+            const options = { headers };
+
+            if ( isRangeReq )
+            {
+                options.range = { start, end };
+            }
+
+            const data = await app.webFetch( link, options );
+
+            if ( headers.range )
```

Comment:
I think this condition is wrong, since when we receive `bytes=0-` we are not passing the `options` so we are then returning
`Content-Range: undefined` and `Content-Length: undefined `. It seems the condition should be `if (isRangeReq) {` instead.